src/main/java/org/jabref/model/pdf/FileAnnotation.java

```diff
@@ -1,127 +1,116 @@
         this.author = dict.getString(COSName.T);
-        this.commentId = annotation.getAnnotationName();
-        this.date = annotation.getModifiedDate();
-        this.page = page;
+        this.date = prettyPrint(annotation.getModifiedDate());
+        this.page = pageNumber;
         this.content = annotation.getContents();
         this.annotationType = annotation.getSubtype();
-
+        this.linkedFileAnnotation = Optional.empty();
     }
 
     /**
-     * Abbreviate annotation names when they are longer than {@code ABBREVIATED_ANNOTATION_NAME_LENGTH} chars
-     * @param annotationName
-     * @return the abbreviated name
+     * For creating a FileAnnotation that has a connection to another FileAnnotation. Needed when creating a text
+     * highlight annotation with a sticky note.
+     *
+     * @param annotation           The actual annotation that holds the information
+     * @param pageNumber           The page of the pdf where the annotation occurs
+     * @param linkedFileAnnotation The corresponding note of a highlighted text area.
      */
-    private String abbreviateAnnotationName(final String annotationName ){
-
-            int abbreviatedContentLengthForName = ABBREVIATED_ANNOTATION_NAME_LENGTH;
-            if (annotationName.length() > abbreviatedContentLengthForName) {
-                return annotationName.subSequence(0, abbreviatedContentLengthForName).toString() + "...";
-            }
-        return annotationName;
-    }
-
-    public void linkComments(FileAnnotation commentToLinkTo){
-        linkedFileAnnotation = commentToLinkTo;
-        commentToLinkTo.setLinkedFileAnnotation(this);
-        commentToLinkTo.setLinkedAnnotation(true);
-        linkedAnnotation = true;
-    }
-
-    @Override
-    public String toString() {
-        return abbreviateAnnotationName(content);
-    }
-
-    public FileAnnotation getLinkedFileAnnotation() {
-        return linkedFileAnnotation;
-    }
-
-    public void setLinkedFileAnnotation(FileAnnotation linkedFileAnnotation) {
-        this.linkedFileAnnotation = linkedFileAnnotation;
-    }
-
-    public String getCommentId() {
-        return commentId;
-    }
-
-    public void setCommentId(String commentId) {
-        this.commentId = commentId;
+    public FileAnnotation(final PDAnnotation annotation, final int pageNumber, FileAnnotation linkedFileAnnotation) {
+        COSDictionary dict = annotation.getDictionary();
+        this.author = dict.getString(COSName.T);
+        this.date = prettyPrint(annotation.getModifiedDate());
+        this.page = pageNumber;
+        this.content = annotation.getContents();
+        this.annotationType = annotation.getSubtype();
+        this.linkedFileAnnotation = Optional.of(linkedFileAnnotation);
     }
 
-    public String getAuthor() {
-        return author;
-    }
+    private String prettyPrint(String date) {
+        // Sometimes this is null, not sure why.
+        if (date == null) {
+            return date;
```

Comment:
Then better set date here to the empty string (otherwise the null-check has to propagate to every usage of  the field `FileAnnotation.date`